l1/eth_subscriber.go

```diff
@@ -0,0 +1,42 @@
+package l1
+
+import (
+	"context"
+
+	"github.com/NethermindEth/juno/l1/internal/contract"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/event"
+)
+
+type EthSubscriber struct {
+	ethClient *ethclient.Client
+	filterer  *contract.StarknetFilterer
+}
+
+var _ Subscriber = (*EthSubscriber)(nil)
+
+func NewEthSubscriber(ethClientAddress string, coreContractAddress common.Address) (*EthSubscriber, error) {
+	ethClient, err := ethclient.Dial(ethClientAddress)
```

Comment:
There is a `ethClient.Close()` method which is not used atm. I think we should call it during shutdown, but maybe it's not critical